zndraw/static/World/components/particles.js

```diff
@@ -252,6 +252,13 @@ class ParticlesGroup extends THREE.Group {
       }
     }
   }
+
+  get_center() {
+    const center = new THREE.Vector3();
+    this.particles_mesh.computeBoundingSphere();
+    center.copy(this.particles_mesh.boundingSphere.center);
+    return center;
+  }
```

Comment:
Do you think we could extend this function, such that it gives back the center of the selected, if there are any as well. This would simplify the function in `events.js` drastically, and keep the logic closer to the `ParticlesGroup`.

Reply:
Sure. Is `this.particles` a list? If so, we could create a temporary mesh from the subset of selected particles and the remaining of the code would remain almost the same

Reply:
I think we would have to use `this.particles_mesh.getMatrixAt(index)`  for each selected.
Through the usage of instancing this is a little more complex than I thought, so we can also fix this in another PR and we can merge this one when the other comment is resolved.


see https://threejs.org/docs/#api/en/objects/InstancedMesh

Reply:
I've extend the atoms class in `cache.js` to support indexing and a way to calculate the mean positions of the Atoms object. With this the function in `events.js` becomes super simple